src/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs

```diff
@@ -168,7 +171,19 @@ internal void InternalEmit(OpCode opcode)
 
             m_ILStream[m_length++] = (byte)opcode.Value;
 
-            UpdateStackSize(opcode, opcode.StackChange());
+            // Manual inline from UpdateStackSize(OpCode, int)
```

Comment:
Hmmm this code duplication seems quite ugly - is there a better way to do this?

Reply:
I tried with a helper method to factor the duplicate code out, but even with the AggressiveInlining Attribute it wouldn't inline and had a significant impact on performance. It depends what is more important here, read-/maintainability or performance

Reply:
Have you tried to measure performance with tiering disabled? I may be completely off but I wonder if the reason for not inlining could have been that it was JITed in the fast tier 0 mode with no optimizations.

Reply:
I just tried again and seems like the JIT inlines afterall. I reverted it back.